pkg/cstor/volume/v1alpha1/build.go

```diff
@@ -296,6 +301,27 @@ func (b *Builder) WithReplicationFactor(replicationfactor int) *Builder {
 	return b
 }
 
+// WithCondition overwrites the existing condition with provided argument if
+// condition not present it will append to existing conditions
+func (b *Builder) WithCondition(
+	conditionStatus apis.CStorVolumeCondition,
+	condType apis.CStorVolumeConditionType) *Builder {
+	isCondPresent := false
+	for i, cond := range b.cstorvolume.object.Status.Conditions {
```

Comment:
already a new fn has been added to get conditionstatus.. you may just need to return pointer from that GetConditionStatus function which can be reused here

Reply:
I will go with something like `func GetConditionStatus(args...) (*apis.CStorVolumeCondition, error)` or any suggestions?

Reply:
If I am not wrong this pointer will not point to underlying array address.